docs/src/main/sphinx/functions/json.rst

```diff
@@ -2,6 +2,1152 @@
+
+Let ``<path>`` return a sequence of three JSON arrays:
+
+.. code-block:: text
+
+    [0, 1, 2], ["a", "b", "c", "d"], [null, null]
+    <path>.size() --> 3, 4, 2
+
+Limitations
+^^^^^^^^^^^
+
+The SQL standard describes the ``datetime()`` JSON path item method and the
+``like_regex()`` JSON path predicate. Trino does not support them yet.
+
+.. _path_modes:
+
+JSON path modes
+^^^^^^^^^^^^^^^
+
+The JSON path expression can be evaluated in two modes: ``strict`` and ``lax``.
+In the ``strict`` mode, it is required that the input JSON data strictly fits
+the schema required by the path expression. In the ``lax`` mode, the input JSON
+data can diverge from the expected schema in certain ways.
+
+The following table shows the differences between the two modes.
+
+.. list-table::
+   :widths: 40 20 40
+   :header-rows: 1
+
+   * - Condition
+     - strict mode
+     - lax mode
+   * - Performing an operation which requires a non-array on an array, e.g.:
+
+       ``$.key`` requires a JSON object
+
+       ``$.floor()`` requires a numeric value
+     - ERROR
+     - The array is automatically unnested, and the operation is performed on
+       each array element.
+   * - Performing an operation which requires an array on an non-array, e.g.:
+
+       ``$[0]``, ``$[*]``, ``$.size()``
+     - ERROR
+     - The non-array item is automatically wrapped in a singleton array, and
+       the operation is performed on the array.
+   * - A structural error: accessing a non-existent element of an array or a
+       non-existent member of a JSON object, e.g.:
+
+       ``$[-1]`` (array index out of bounds)
+
+       ``$.key``, where the input JSON object does not have a member ``key``
+     - ERROR
+     - The error is suppressed, and the operation results in an empty sequence.
+
+Examples of the lax mode behavior:
+
+Let ``<path>`` return a sequence of three items, a JSON array, a JSON object,
+and a scalar numeric value:
+
+.. code-block:: text
+
+    [1, "a", null], {"key1" : 1.0, "key2" : true}, -2e3
+
+The following example shows the wildcard array accessor in the lax mode. The
+JSON array returns all its elements, while the JSON object and the number are
+wrapped in singleton arrays and then unnested, so effectively they appear
+unchanged in the output sequence:
+
+.. code-block:: text
+
+    <path>[*] --> 1, "a", null, {"key1" : 1.0, "key2" : true}, -2e3
+
+When calling the ``size()`` method, the JSON object and the number are also
+wrapped in singleton arrays:
+
+.. code-block:: text
+
+    <path>.size() --> 3, 1, 1
+
+In some cases, the lax mode cannot prevent failure. Even though the JSON array
+is unwrapped prior to calling the ``floor()`` method, the item ``"a"`` causes
+type mismatch.
+
+.. code-block:: text
+
+    <path>.floor() --> ERROR
+
+.. _json_exists:
+
+JSON_EXISTS
+-----------
+
+The ``json_exists`` function determines whether a JSON value satisfies a JSON
+path specification.
+
+.. code-block:: text
+
+    JSON_EXISTS(
```

Comment:
lowercase? 


this is going to be tricky since its so complex .. lets to whatever we decided for other complex functions

Reply:
I'm leaving it in uppercase in synopsis, and changing to lowercase in examples.